### PR TITLE
04 connect to api

### PR DESCRIPTION
**Pull request by FaranoshAmini**
In this pull request, I have implemented : 

 - Used fetch for API calls
 - Used thunk
 - Display book list by fetching data from API
 - Update Action Creator of Add and Remove so that returns the function

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -39,20 +27,77 @@ const booksReducer = (state = initialState, action) => {
...
+        title: book.title,
+        author: book.author,
+        category: book.category,
+      }),
+      headers: { 'Content-type': 'application/json; charset=UTF-8' },
+    });
+    if (response.status === 201) {
+      dispatch({
+        type: ADD_BOOK,
+        payload: book,
```
Comment (youmari) [inline src/redux/books/books.js:74]: - [ ] as you can see in the screenshot I'm unable to get the books or create a book and also there are bunch of 404 errors in the console returned back from the API so please make sure to fix that in order your app is working properly without any errors 
Screen Shot 2022-11-11 at 17 16 29

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @FaranoshAmini , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @FaranoshAmini, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by FaranoshAmini** [comment]
@Ismailco Thanks for your kind review.

**Pull request merged**